pre_commit/languages/coursier.py

```diff
@@ -27,19 +28,21 @@ def install_environment(
     helpers.assert_no_additional_deps('coursier', additional_dependencies)
 
     envdir = prefix.path(helpers.environment_dir(ENVIRONMENT_DIR, version))
+    channel = prefix.path('.pre-commit-channel')
     with clean_path_on_failure(envdir):
-        os.mkdir(envdir)
-        helpers.run_setup_cmd(
-            prefix,
-            (
-                'coursier',
-                'install',
-                '--default-channels=false',
-                '--channel=.pre-commit-channel',
-                'pre-commit-hook',
-                f'--dir={envdir}',
-            ),
-        )
+        for app_descriptor in os.listdir(channel):
+            app = Path(app_descriptor).stem
```

Comment:
```python
_, app = os.path.split(app_descriptor)
```

Reply:
I also want to trim the file extension because thats what coursier expects. This doesn't appear to do that.

```python
>>> import os
>>> _, app = os.path.split('/Users/jmoniz/src/scala-hooks/.precommit-channel/scalafmt.json')
>>> assert app == 'scalafmt'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
```

The existing code does what it needs to:
```python
>>> from pathlib import Path
>>> app = Path('/Users/jmoniz/src/scala-hooks/.precommit-channel/scalafmt.json').stem
>>> assert app == 'scalafmt'
```

Is there a simpler way to get the file base name?

Reply:
os.path.split + os.path.splitext

Reply:
Nice! the following works perfectly:
```python
>>> import os
>>> path = '/Users/jmoniz/src/scala-hooks/.precommit-channel/scalafmt.json'
>>> _, app_file = os.path.split(path)
>>> app, _ = os.path.splitext(app_file)
>>> assert app == 'scalafmt'
>>>
```
I'll go ahead and make this change.